### PR TITLE
New Encoder Lib, some bugfixes

### PR DESCRIPTION
**Pull request by Williangalvani**

### CODE REVIEW

File: Firmware/ControladorRobo/ControladorRobo.ino [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #include "irarray.h"
 #include "hbridge.h"
 #include "controle.h"
-#include "encoder.h"
+#include "Encoder/Encoder.h"
```
Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino]: Move submodules to a /lib

File: Firmware/ControladorRobo/ControladorRobo.ino [hunk 2]
```diff
@@ -81,23 +80,23 @@ ISR(TIMER1_COMPA_vect)
...
 
-    wEsq = arcLeft*freq;
-    wDir = arcRight*freq;
+    float rightAngularSpeed = rightAngularDelta*freq;
+    float leftAngularSpeed = leftAngularDelta*freq;
 
-    posDir+=arcRight;
-    posEsq+=arcLeft;
+    rightAngPos += rightAngularDelta;
+    leftAngPos += leftAngularDelta;
```
Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino:93]: Add int with float ?

File: Firmware/ControladorRobo/ControladorRobo.ino [hunk 3]
```diff
@@ -81,23 +80,23 @@ ISR(TIMER1_COMPA_vect)
...
 
-    pulsosDir = 0;
-    pulsosEsq = 0;
+    left.write(0);
+    right.write(0);
 
-    float u_esq = controle_esq.update(refEsq, wEsq);
-    float u_dir = controle_dir.update(refDir, wDir);
+    float u_esq = controle_esq.update(refEsq, leftAngularSpeed);
+    float u_dir = controle_dir.update(refDir, rightAngularSpeed);
```
Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino]: *AngularSpeed doesn't make sense to be long if the value is converted to float in Controller::update

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 2 above)
Comment (Williangalvani) [inline Firmware/ControladorRobo/ControladorRobo.ino:93]: What type do you suggest there? float?

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 3 above)
Comment (Williangalvani) [inline Firmware/ControladorRobo/ControladorRobo.ino]: They are float, defined in lines 88 and 89

File: Firmware/ControladorRobo/ControladorRobo.ino [hunk 4]
```diff
@@ -46,8 +47,6 @@ void setup()
 
     HBridge::self();
```
Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino]: Why is this here ?

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 3 above)
Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino]: Sorry, mistook  with *AngPos.

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 2 above)
Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino:93]: > Long variables are extended size variables for number storage, and store 32 bits (4 bytes), from -2,147,483,648 to 2,147,483,647.
> 
> Floating-point numbers can be as large as 3.4028235E+38 and as low as -3.4028235E+38. They are stored as 32 bits (4 bytes) of information.

Move to float.

File: Firmware/ControladorRobo/ControladorRobo.ino [hunk 5]
```diff
@@ -81,23 +80,23 @@ ISR(TIMER1_COMPA_vect)
...
-    float u_esq = controle_esq.update(refEsq, wEsq);
-    float u_dir = controle_dir.update(refDir, wDir);
+    float u_esq = controle_esq.update(refEsq, leftAngularSpeed);
+    float u_dir = controle_dir.update(refDir, rightAngularSpeed);
 
     HBridge* bridge = &HBridge::self();
-    bridge->setWheelPWM(bridge->LEFT,(int)floor(u_dir*51));
-    bridge->setWheelPWM(bridge->RIGHT,(int)floor(u_esq*51));
+    bridge->setWheelPWM(bridge->LEFT,(int)floor(u_esq*51));
+    bridge->setWheelPWM(bridge->RIGHT,(int)floor(u_dir*51));
```
Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino:103]: Why 51 ?

Comment (Ghizoni) [inline Firmware/ControladorRobo/ControladorRobo.ino:103]: 51 is related with 255 from the PWM and 5.0 V.
^^

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 4 above)
Comment (Williangalvani) [inline Firmware/ControladorRobo/ControladorRobo.ino]: Because the encoder.h no longer exists, and I didn't think there was enough content to justify it's existence.
Suggestions?

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 5 above)
Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino:103]: Ok, So u_\* have a max value of 5 ?

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 4 above)
Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino]: Can you have a more complete answer ?
I can't see why are you calling the HBridge::self(), you can't delete a singleton if it's what are you trying to do. Or: The initialization begun in the first call, don't need to worry about that.

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 1 above)
Comment (Williangalvani) [inline Firmware/ControladorRobo/ControladorRobo.ino]: done

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 2 above)
Comment (Williangalvani) [inline Firmware/ControladorRobo/ControladorRobo.ino:93]: ✔

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 4 above)
Comment (Williangalvani) [inline Firmware/ControladorRobo/ControladorRobo.ino]: I'm just making sure it's initialized on the beginning of the code, to avoid the H-Bridge pins being initliazed only when we try to move.

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 5 above)
Comment (Williangalvani) [inline Firmware/ControladorRobo/ControladorRobo.ino:103]: This turns the 0-5 controller output into 0-255 number for the H-bridge.
But it's a "xunxo" and needs to be done properly

Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino:103]: So, the maximum robot's velocity is 5 m/s | Km/h | pc/tP ?
ps: The control output has been discovered that is in volts, this value is multiplied by 51 to be inside the 255 range value of PWM.

Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino:103]: Talking with @Williangalvani, the best  solution that we saw to this is passing the control's output value to HBridge that will deal with the Voltage/Actuation conversion.

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 3 above)
Comment (Williangalvani) [inline Firmware/ControladorRobo/ControladorRobo.ino]: ✔

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 4 above)
Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino]: Add a comment about that in code.

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 1 above)
Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino]: #include "lib/Encoder/Encoder.h"

File: Firmware/ControladorRobo/ControladorRobo.ino [hunk 6]
```diff
@@ -4,7 +4,7 @@
 #include "irarray.h"
 #include "hbridge.h"
 #include "controle.h"
-#include "encoder.h"
+#include "lib/Encoder/Encoder.h"
```
Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino]: You need to correct this in 	ccd46cc

File: Firmware/ControladorRobo/ControladorRobo.ino (hunk 2 above)
Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino:93]: Where is this used ?

Comment (Williangalvani) [inline Firmware/ControladorRobo/ControladorRobo.ino:93]: it's not yet. but I intend to use the final angular or linear position to measure rotation angles.

Comment (patrickelectric) [inline Firmware/ControladorRobo/ControladorRobo.ino:93]: Ok, but comment that in code.
